### ISSUE TITLE
SSL hangs at socketFactory.createSocket()

### ISSUE THREAD

**Issue by lyonzy** [NathanaelA/nativescript-websockets#5]
Hello,

I'm trying to use this to connect to Firebase, which uses wss://, and for some reason it's getting stuck at the socketFactory.createSocket() call. I've separated line 166 of websockets.android.js to:
```javascript
        var s = socketFactory.createSocket();
        this._socket.setSocket( s );
```

When I debug it, I can't step past that first call. Any idea what might be happening?

I'm going to try to reproduce this in Java and if that works it must be a Nativescript issue. 

One possible cause I've thought of is the use of the parameterless createSocket() call - in some situations that throws a "Unconnected sockets not implemented" (see [here](http://grepcode.com/file/repo1.maven.org/maven2/org.robolectric/android-all/5.0.0_r2-robolectric-1/javax/net/SocketFactory.java#SocketFactory.createSocket%28%29)) so maybe that's what's happening?

**Comment by lyonzy** [comment]
Nope, weird... the equivalent code works flawlessly in a blank Android Studio app:

```Java
javax.net.ssl.SSLContext sslContext = javax.net.ssl.SSLContext.getInstance("TLS");
sslContext.init(null, null, null);
javax.net.ssl.SSLSocketFactory socketFactory = sslContext.getSocketFactory();
javax.net.ssl.SSLSocket s = (SSLSocket) socketFactory.createSocket();
```
![image](https://cloud.githubusercontent.com/assets/11625419/11326639/83fcad8c-91c2-11e5-9e12-68e1fe3d4534.png)

**Comment by lyonzy** [comment]
Ok, so it looks like it's happening somewhere in Nativescript while trying to convert the SSLSocket to a JavaScript object. Here's what I've got working:
```Java
public class SSLSocketHelper {

    public static SSLSocket createSocket() throws KeyManagementException, NoSuchAlgorithmException, IOException {

        javax.net.ssl.SSLContext sslContext = javax.net.ssl.SSLContext.getInstance("TLS");
        sslContext.init(null, null, null);

        javax.net.ssl.SSLSocketFactory socketFactory = sslContext.getSocketFactory();

        javax.net.ssl.SSLSocket s = (SSLSocket) socketFactory.createSocket();

        return s;
    }

    public static void addSocket(org.java_websocket.client.WebSocketClient client) throws KeyManagementException, NoSuchAlgorithmException, IOException {
        client.setSocket(SSLSocketHelper.createSocket());
    }
}
```
```Javascript
com.lyonzy.sslsockethelper.SSLSocketHelper.addSocket(this._socket);
```

Exactly why Nativescript is struggling to convert the type though, I'm not sure...

**Comment by NathanaelA** [comment]
It is possible I haven't messed with the WSS stuff in ages.  NativeScript has changed a lot since then.  Which version of NativeScript?   Does normal ws work for you?

**Comment by NathanaelA** [comment]
Ok, I duplicated the issue.  And you are right it is something messed up in the android runtimes.  I have worked around the issue pretty much the same way you did by embedding the ssl creation code inside the websocket library.  I have also fixed several other library issues and updated the websocket code to use ios cocoapods so that their is no more manual steps on either ios or android.
v1.10 fixes this and other issues.  :grinning:
